robottelo/helpers.py

```diff
@@ -716,3 +716,64 @@ def idgen(val):
     test name
     """
     return '_parameter'
+
+
+class InstallerCommand:
+    """This class constructs, parses, updates and gets formatted installer commands"""
+
+    def __init__(self, *args, command='satellite-installer', allow_dupes=False, **kwargs):
+        self.command = command
+        self.args = kwargs.pop('installer_args', [])
+        self.opts = kwargs.pop('installer_opts', {})
+        self.update(*args, allow_dupes=allow_dupes, **kwargs)
+
+    def get_command(self):
+        command_str = self.command
+        for arg in self.args:
+            command_str += f' {"-" if len(arg) == 1 else "--"}{arg}'
+        for key, val in self.opts.items():
+            if isinstance(val, list):
+                for v in val:
+                    command_str += f' --{key.replace("_", "-")} {v}'
+            else:
+                command_str += f' --{key.replace("_", "-")} {val}'
+        return command_str
+
+    def update(self, *args, allow_dupes=False, **kwargs):
+        new_args = [arg for arg in args if arg not in self.args]
+        self.args.extend(new_args)
+        if not allow_dupes:
+            self.opts.update(kwargs)
+        for key, val in kwargs.items():
+            if not allow_dupes:
+                self.opts[key] = val
+            elif (curr_val := self.opts.get(key)) :  # noqa: E203
+                val = [val]
+                if not isinstance(curr_val, list):
+                    curr_val = [curr_val]
+                val += curr_val
+            self.opts[key] = val
+
+    @classmethod
+    def from_cmd_str(cls, command='satellite-installer', cmd_str=None):
```

Comment:
is this only used to for certain scenarios ? can be used anywhere. can you add example how can this used in doc string something.

Reply:
it can theoretically be used for any scenario/installer. well technically any command..